apps/examples/testcase/le_tc/network/itc_net_connect.c

```diff
@@ -0,0 +1,253 @@
+{
+	int server_socket;
+	int client_socket;
+	int valread;
+	int opt = 1;
+	int ret;
+	struct sockaddr_in address;
+	int addrlen = sizeof(address);
+	int num_clients = *((int *)ptr_num_clients);
+	char buffer[BUFF_LEN] = {0};
+	char *ptr_msg = SERVER_MSG;
+
+	server_socket = socket(AF_INET, SOCK_STREAM, 0);
+	if (server_socket < 0) {
+		return NULL;
+	}
+	setsockopt(server_socket, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT, &opt, sizeof(opt));
+	address.sin_family = AF_INET;
+	address.sin_addr.s_addr = INADDR_ANY;
+	address.sin_port = htons(g_port);
+	ret = bind(server_socket, (struct sockaddr *)&address, sizeof(address));
+	if (ret != OK) {
+		close(server_socket);
+		return NULL;
+	}
+	ret = listen(server_socket, 5);
+	if (ret != OK) {
+		close(server_socket);
+		return NULL;
+	}
+	while (num_clients-- > 0) {
+		client_socket = accept(server_socket, (struct sockaddr *)&address, (socklen_t*)&addrlen);
+		if (client_socket < 0) {
+			close(server_socket);
+			return NULL;
+		}
+		memset(buffer, 0, BUFF_LEN);
+		valread = read(client_socket, buffer, BUFF_LEN);
+		if (valread < 0) {
+			close(client_socket);
+			close(server_socket);
+			return NULL;
+		}
+		strncpy(buffer + valread, ptr_msg, sizeof(SERVER_MSG));
+		ret = send(client_socket, buffer, strlen(buffer), 0);
+		if (ret != strlen(CLIENT_MSG) + strlen(SERVER_MSG)) {
+			close(client_socket);
+			close(server_socket);
+			return NULL;
+		}
+		sleep(WAIT_TIME_SEC_2);
+		ret = close(client_socket);
+		if (ret != OK) {
+			close(server_socket);
+			return NULL;
+		}
+	}
+	ret = close(server_socket);
+	if (ret != OK) {
+		return NULL;
+	}
+	return NULL;
+}
+
+/**
+* @fn               :client_connect
+* @description      :starts client
+* @return           :void
+*/
+static void *client_connect(void *ptr_id)
+{
+	int *id;
+	id = (int *)ptr_id;
+	int sock = 0;
+	int ret;
+	int valread;
+	struct sockaddr_in serv_addr;
+	char client_msg[BUFF_LEN];
+	char buffer[BUFF_LEN] = {0};
+
+	sock = socket(AF_INET, SOCK_STREAM, 0);
+	if (sock < 0) {
+		return NULL;
+	}
+	memset(&serv_addr, '0', sizeof(serv_addr));
+	serv_addr.sin_family = AF_INET;
+	serv_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	serv_addr.sin_port = htons(g_port);
+	ret = connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr));
+	if (ret != OK) {
+		close(sock);
+		return NULL;
+	}
+	strncpy(client_msg, CLIENT_MSG, sizeof(CLIENT_MSG));
+	client_msg[strlen(client_msg) - 2] = '0' + (*id);
+	ret = send(sock, client_msg, strlen(client_msg), 0);
+	if (ret != strlen(client_msg)) {
+		close(sock);
+		return NULL;
+	}
+	valread = read(sock, buffer, BUFF_LEN);
+	if (valread != strlen(client_msg) + strlen(SERVER_MSG)) {
+		close(sock);
+		return NULL;
+	}
+	if (strncmp(buffer, client_msg, strlen(client_msg)) != 0) {
+		close(sock);
+		return NULL;
+	}
+	if (strncmp(buffer + strlen(client_msg), SERVER_MSG, strlen(SERVER_MSG)) != 0) {
+		close(sock);
+		return NULL;
+	}
+	sleep(WAIT_TIME_SEC_2);
```

Comment:
the reason why I fix SEC_2 to WAIT_TIME is modifiability. if we change WAIT_TIME_SEC_2 to 3 seconds than we need to change WAIT_TIME_SEC_2 too.

Reply:
Fixed review comments